src/bin/hue_register_user.rs

```diff
@@ -9,14 +9,14 @@ fn main() {
     if args.len() != 2 {
         println!("usage : {:?} <devicetype>", args[0]);
     } else {
-        let mut bridge = ::hueclient::bridge::Bridge::discover_required();
+        let bridge = ::hueclient::Bridge::discover_required();
         println!("posting user {:?} in {:?}", args[1], bridge);
-        while true {
-            let r = bridge.register_user(&args[1]);
+        loop {
+            let r = bridge.clone().register_user(&args[1]);
```

Comment:
Ha yeah, so we need this clone() all of a sudden... Should the new API return the old bridge in the Error case instead ?

Reply:
Yeah this is true. I'd personally don't like returning the UnauthBridge in the error case, but the problem is that we cannot say "compiler, please move `self` if the function succeeds, but leave it intact if the function fails. We could create a struct `BorrowedBridge`, which borrows the password and client from the UnauthBridge, but I doubt that that is worth the effort.

Reply:
OK, to be fair, the clone() is needed in this code because we perform a retry loop around the register_user. We don't really expect to do this in "normal" initialisation code (because we will just fail and drop the old bridge) , so we'll assume it is ok like this.